lib/tlscontext.c

```diff
@@ -602,15 +602,15 @@ _are_key_and_cert_files_accessible(TLSContext *self)
 }
 
 static gboolean
-_client_key_and_cert_files_are_not_specified(TLSContext *self)
+_key_and_cert_files_are_not_specified(TLSContext *self)
 {
-  return self->mode == TM_CLIENT && (!self->key_file && !self->cert_file);
```

Comment:
Omitting the key in server mode should not be allowed in my opinion.
There is no secure cipher in OpenSSL that works without a key, so every connection attempt will fail with an SSL error: "no shared cipher".

I understand that this is required to fix `default-network-drivers` (since we can't conditionally remove a source from the config based on the value of a SCL block parameter). Adding such a feature to the grammar would be a much bigger job, so I can reconcile with this patch :smile: .